File: common/src/main/java/com/pluxity/facility/category/FacilityCategoryRepository.java
package com.pluxity.facility.category;

import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface FacilityCategoryRepository extends JpaRepository<FacilityCategory, Long> {
    Optional<FacilityCategory> findByNameAndParentId(String name, Long parentId);

    List<FacilityCategory> findAllByParentIsNull();
}

File: common/src/main/kotlin/com/pluxity/facility/category/FacilityCategoryRepository.kt
package com.pluxity.facility.category

import org.springframework.data.domain.Sort
import org.springframework.data.jpa.repository.JpaRepository

interface FacilityCategoryRepository : JpaRepository<FacilityCategory, Long> {
    fun findByNameAndParentId(
        name: String,
        parentId: Long?,
    ): FacilityCategory?

    fun findByParentIsNull(sort: Sort): List<FacilityCategory>
}
